contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -502,70 +444,56 @@ contract RibbonTreasuryVault is
 
     /**
      * @notice Adds new whitelisted address
-     * @param newWhitelist is a list of address to include to the whitelist
+     * @param newWhitelist is the address to include in the whitelist
      */
-    function addWhitelist(address[] calldata newWhitelist)
+    function addWhitelist(address newWhitelist)
         external
-        onlyKeeper
+        onlyOwner
         nonReentrant
     {
+        _addWhitelist(newWhitelist);
+    }
+
+    /**
+     * @notice Internal function to add new whitelisted address
+     * @param newWhitelist is the address to include in the whitelist
+     */
+    function _addWhitelist(address newWhitelist) internal {
+        require(newWhitelist != address(0), "Whitelist null");
+        require(!whitelistMap[newWhitelist], "Whitelist duplicate");
         require(
-            (newWhitelist.length + whitelistArray.length) <= WHITELIST_LIMIT,
+            (whitelistArray.length + 1) <= WHITELIST_LIMIT,
             "Whitelist exceed limit"
         );
 
-        for (uint256 i = 0; i < newWhitelist.length; i++) {
-            require(newWhitelist[i] != address(0), "Whitelist null");
-            require(!whitelistMap[newWhitelist[i]], "Whitelist duplicate");
-
-            whitelistMap[newWhitelist[i]] = true;
-            whitelistArray.push(newWhitelist[i]);
-        }
+        whitelistMap[newWhitelist] = true;
+        whitelistArray.push(newWhitelist);
     }
 
     /**
      * @notice Remove addresses from whitelist
-     * @param excludeWhitelist is a list of address to include to the whitelist
+     * @param excludeWhitelist is the address to exclude from the whitelist
      */
-    function removeWhitelist(address[] calldata excludeWhitelist)
+    function removeWhitelist(address excludeWhitelist)
         external
-        onlyKeeper
+        onlyOwner
         nonReentrant
     {
-        require(
-            (whitelistArray.length - excludeWhitelist.length) > 0,
-            "Whitelist cannot be empty"
-        );
+        require(whitelistMap[excludeWhitelist], "Whitelist does not exist");
+        require((whitelistArray.length - 1) > 0, "Whitelist cannot be empty");
```

Comment:
Accessing `whitelistArray.length` takes up 220 gas each time, so we can save that lookup by using a variable

```solidity
uint whitelistLength = whitelistArray.length;

require((whitelistLength - 1) > 0, "Whitelist cannot be empty");

for (uint256 i = 0; i < whitelistLength; i++) {
  
}
```

Reply:
Also shouldn't the require statement be:

```solidity
require(whitelistLength > 0, "Whitelist cannot be empty");
```

You don't need to do a `- 1`.

Reply:
A better way to write this statement is:

```
require(whitelistLength >= 1, "Whitelist cannot be empty");
```

In the edge case that `whitelistLength = 0` you do `0 - 1` and it will revert